pkg/apis/pingcap/v1alpha1/tidbcluster.go

```diff
@@ -59,16 +59,24 @@ func (tc *TidbCluster) PDAutoFailovering() bool {
 	return false
 }
 
-func (tc *TidbCluster) PDRealReplicas() int32 {
+func (tc *TidbCluster) PDDesiredReplicas() int32 {
```

Comment:
I prefer to rename `<TiDB/PD/TiKV>RealReplicas` to `<TiDB/PD/TiKV>DesiredReplicas`, because `tc.<TiDB/PD/TiKV>.Replicas + tc.Status.<TiDB/PD/TiKV>.FailureMembers` is the desired replicas.

Reply:
I think this name should not be modified. The desired replicas represent the replicas set by the user. When a failover occurs, the actual replicas will be higher than the user settings.

Reply:
len(failoverMembers) is only increased when the user enabled auto-failover, so the desired replicas is what the user want

Reply:
> I think this name should not be modified. The desired replicas represent the replicas set by the user. When a failover occurs, the actual replicas will be higher than the user settings.

I agreed so.

Reply:
as suggested by @onlymellb , I've added statefulset prefix (`sts`) to indicate that these are desired/actual replicas of stateful sets we are managing.
cc @weekface